core/proxy.c

```diff
@@ -397,6 +406,94 @@ oio_proxy_call_container_get_properties (CURL *h, struct oio_url_s *u,
 }
 
 GError *
+oio_proxy_call_content_get_properties (CURL *h, struct oio_url_s *u,
+					 GString **props_str)
+{
+	GString *http_url = _curl_content_url (u, "get_properties");
+	struct http_ctx_s out_ctx = {0};
+	if (props_str && *props_str)
+		out_ctx.body = *props_str;
+	else
+		out_ctx.body = g_string_sized_new(512);
+
+	GError *err = _proxy_call (h, "POST", http_url->str, NULL, &out_ctx);
+
+	g_string_free(http_url, TRUE);
+	if (props_str && !*props_str && !err)
+		*props_str = out_ctx.body;
+	else
+		g_string_free(out_ctx.body, TRUE);
+	return err;
+}
+
+GError *
+oio_proxy_call_container_set_properties (CURL *h, struct oio_url_s *u,
+				       const char* const *values) {
+	int cmp = 0;
+	GError *err = NULL;
+	gchar *tmp = NULL;
+	gchar *result = NULL;
+	GString *val = NULL;
+	for (cmp = 0; NULL != *(values+cmp) && NULL != *(values+cmp+1)
+		     ; cmp+=2 )
+		{
+			if(cmp != 0) {
+				result = g_strdup_printf("%s,\"%s\":\"%s\"",tmp,values [cmp], values [cmp+1]);
+				g_free(tmp);
+				tmp = result;
+			}
+			else
+				tmp = g_strdup_printf("{\"%s\":\"%s\"",values [cmp], values [cmp+1]);
+		}
+	val = _gs_vprintf("%s}", result);
+	g_free(result);
+	struct http_ctx_s i = {
+		                .body = val
+	};
+	GString *http_url = _curl_container_url (u, "set_properties");
+	err = _proxy_call (h, "POST", http_url->str, &i, NULL);
+	g_string_free (http_url, TRUE);
+	g_string_free (i.body, TRUE);
+	if(err)
+		return err;
+	return err;
+}
+
+
+GError *
+oio_proxy_call_content_set_properties (CURL *h, struct oio_url_s *u,
+				       const char* const *values) {
+	int cmp = 0;
+	GError *err = NULL;
+	gchar *tmp = NULL;
+	gchar *result = NULL;
+	GString *val = NULL;
+	for (cmp = 0; NULL != *(values+cmp) && NULL != *(values+cmp+1)
```

Comment:
* `cmp` is unused out of the loop. We write C99, and you can declare the variable in the `for` statement.
* `cmp` seems to be just an index. A more intuitive name is required here. Why not just `i` ?
* `*(values+cmp+1)` can be even shortened to `values[i+1]`

```
for (int i=0; values[i] && values[i+1] ;i+=2) {
```
instead of
```
for (cmp = 0; NULL != *(values+cmp) && NULL != *(values+cmp+1) ; cmp+=2 ) {
```